components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -136,16 +136,23 @@ NTPSponsoredImagesData* ViewCounterService::GetCurrentBrandedWallpaperData()
 }
 
 base::Value ViewCounterService::GetCurrentWallpaperForDisplay() {
-  if (ShouldShowBrandedWallpaper()) {
-    base::Value branded_wallpaper = GetCurrentBrandedWallpaper();
+  if (IsBrandedWallpaperActive()) {
+    base::Value branded_wallpaper;
+    if (model_.ShouldShowBrandedWallpaper()) {
+      branded_wallpaper = GetCurrentBrandedWallpaper();
+      if (!branded_wallpaper.is_dict()) {
+        // Failed to get branded wallpaper as it was frequency capped by ads
+        // service. In this case next background wallpaper should be shown on
+        // NTP. To do this we need to increment background wallpaper index as it
+        // wasn't incremented during the last RegisterPageView() call.
+        model_.IncreaseBackgroundWallpaperImageIndex();
+      }
+    }
+
+    MaybePrefetchNewTabPageAd();
```

Comment:
Although this logic is not added in this PR, I think this seems error-prone.
Maybe we assume that this method is called once per NTP loading?
However, it seems this method could be called multiple times easily easily as this method name is getter.

How about handling(`increase bg wallpaper index`) this from `RegisterPageView()` instead of here?

Reply:
I will move this logic to RegisterPageView()

Reply:
Yup! 👍🏼